lib/senkyoshi/models/question.rb

```diff
@@ -125,7 +125,7 @@ def set_correct_answers(resprocessing)
         if score_number > 0
           @correct_answers["fraction"] = score_number.to_f / @max_score.to_f
         else
-          @correct_answers["fraction"] = 0
+          @correct_answers["fraction"] = 1
```

Comment:
Blackboard allows for the questions to not be worth points when the question is contained in a survey. So this if else tells canvas that even if a problem isn't worth points it's still correct by marking it a 1. The 1 to canvas is correct and 0 is incorrect. Then any number in between is a percentage of the problem they will get correct if they answer it correctly which applies to like matching questions in which the student needs to get all the matching correct to get the full points. I'll make a comment for this too.